hoomd/md/EvaluatorTersoff.h

```diff
@@ -91,15 +91,15 @@ class EvaluatorTersoff
             pybind11::dict v;
 
             pybind11::list mags;
-            mags.append(coeffs.x);
-            mags.append(coeffs.y);
+            mags.append((Scalar)coeffs.x);
+            mags.append((Scalar)coeffs.y);
```

Comment:
Seems unrelated to AMD compilation. Also, use `static_cast`.

Reply:
If I recall correctly, AMD's HIP uses non-standard types for `float4` and `double4` (and `Scalar4` by proxy) - so this may be necessary. @jglaser can confirm.